### PR TITLE
feat: Add contributing guide & code of conduct

### PR DESCRIPTION
**Pull request by Milouu**
Signed-off-by: Milouu <milan.roustan@owkin.com>

Add a contributing guide & code of conduct to the documentation. We add these in the documentation to centralize it. We will then be able to create CONTRIBUTING.md & CODE_OF_CONDUCT.md files in the other repositories referencing these pages.

### CODE REVIEW

**Review by ThibaultFy — COMMENTED** [review]
To pass the sphinx build, you must resolve every warning and error (click on details on the ci to see them).
Especially, Sphinx is highly dependent on indentation, space and blank lines.
Also, if you want to add a clickable link, you must use this format : 
```
`Owkin` <https://owkin.com/>`_
```

File: docs/source/contributing/contributing-guide.rst
Comment (ThibaultFy) [inline docs/source/contributing/contributing-guide.rst:38]:
```suggestion
For compliance purposes, `Developer Certificate of Origin (DCO) on Pull Requests <https://github.com/apps/dco>`_ is activated on the repo.
```

**Review by ThibaultFy — COMMENTED** [review]
Some fix proposal (to test). You can build the doc locally to debug by launching `make clean && make htlm` from the docs folder.

File: docs/source/contributing/contributing-guide.rst
Comment (ThibaultFy) [inline docs/source/contributing/contributing-guide.rst:25]:
```suggestion
* `**Fork the repository** <https://guides.github.com/activities/forking/>`_ and clone it locally. Connect your local to the original "upstream" repository by adding it as a remote. Pull in changes from "upstream" often so that you stay up to date so that when you submit your pull request, merge conflicts will be less likely. (See more detailed instructions `here <https://help.github.com/articles/syncing-a-fork/>`_.)
```
This typo does not work (see RTD build). Might be working like this (to verify)

File: docs/source/contributing/contributing-guide.rst
Comment (ThibaultFy) [inline docs/source/contributing/contributing-guide.rst:43]: Is this the result we expect ? It seems a bit weird...

File: docs/source/contributing/code-of-conduct.rst
Comment (ThibaultFy) [inline docs/source/contributing/code-of-conduct.rst:79]: Can we just put the url to the 1.4 version instead of having two ?

File: docs/source/contributing/code-of-conduct.rst
Comment (ThibaultFy) [inline docs/source/contributing/code-of-conduct.rst:65]: Does this section is relevant ? I mean with the maintainer github pseudo

File: docs/source/contributing/code-of-conduct.rst
Comment (ThibaultFy) [inline docs/source/contributing/code-of-conduct.rst:30]:
```suggestion
* The use of sexualized language or imagery and unwelcome sexual attention or advances
```

File: docs/source/contributing/code-of-conduct.rst
Comment (ThibaultFy) [inline docs/source/contributing/code-of-conduct.rst:34]:
```suggestion
* Publishing others' private information, such as a physical or electronic address, without explicit permission
```

File: docs/source/contributing/code-of-conduct.rst
Comment (ThibaultFy) [inline docs/source/contributing/code-of-conduct.rst:36]:
```suggestion
* Other conduct which could reasonably be considered inappropriate in a professional setting
```

**Review by ThibaultFy — CHANGES_REQUESTED** [review]
You can check how the build works and render by clicking on the RTD check on the PR.
The comments I made are mainly about the rendering.

File: docs/source/contributing/contributing-guide.rst
Comment (RomainGoussault) [inline docs/source/contributing/contributing-guide.rst:31]: @Milouu we should add something about the contributors file. Every contributor should be mentioned in a CONTRIBUTORS.MD file and then in the LICENCE we mention the substra contributors [here](https://github.com/Substra/substra-documentation/blob/main/LICENSE#L179) instead of Owkin Inc.

File: docs/source/contributing/code-of-conduct.rst
Comment (ThibaultFy) [inline docs/source/contributing/code-of-conduct.rst:62]: Don't we have a mail more specific to this use case ? (asking @RomainGoussault also). Cause this one is quite dependent on the current organisation and might become obsolete quite fast...

**Review by ThibaultFy — COMMENTED** [review]
One small comment on the mail adress, then LGTM.

File: docs/source/contributing/code-of-conduct.rst
Comment (RomainGoussault) [reply docs/source/contributing/code-of-conduct.rst:62]: hum agree, and it should not be an owkin address but a LFAI address.
Another way to put it is to report to the code owners? The email address of the code owner should be accessible?

Comment (ThibaultFy) [reply docs/source/contributing/code-of-conduct.rst:62]: No there no way with github to have this functionnality... I agree for the LFAI address, or maybe ask the IT if we could have a support@substra.com or something like that ? Cause it should be an address we're actually using! And if coming from the IT, we can link the mail to our owkin adress

Comment (RomainGoussault) [reply docs/source/contributing/code-of-conduct.rst:62]: I am checking if we have an LFAI address. I will keep you posted shortly.

Comment (RomainGoussault) [reply docs/source/contributing/code-of-conduct.rst:62]: Sorry I don't have an answer yet. 
Let's not block the merging for this issue though, we will change the address alter.

File: docs/source/contributing/contributing-guide.rst
Comment (RomainGoussault) [inline docs/source/contributing/contributing-guide.rst:11]:
```suggestion
Be kind and thoughtful in your conversations around this project. We all come from different backgrounds and projects, which means we likely have different perspectives on how things should be done. Try to listen to others rather than convince them that your way is correct.

Substra has a :doc:`Contributor Code of Conduct </contributing/code-of-conduct>`. By participating in this project, you agree to abide by its terms.
```

File: docs/source/contributing/contributing-guide.rst
Comment (RomainGoussault) [inline docs/source/contributing/contributing-guide.rst:13]:
```suggestion
Who are contributors?
```

File: docs/source/contributing/contributing-guide.rst
Comment (RomainGoussault) [inline docs/source/contributing/contributing-guide.rst:16]: link is broken
image

File: docs/source/contributing/contributing-guide.rst
Comment (RomainGoussault) [inline docs/source/contributing/contributing-guide.rst:30]:
```suggestion
* `Fork the repository <https://guides.github.com/activities/forking/>`_ and clone it locally. Connect your local to the original "upstream" repository by adding it as a remote. Pull in changes from "upstream" often so that you stay up to date so that when you submit your pull request, merge conflicts will be less likely. See more detailed instructions `here <https://help.github.com/articles/syncing-a-fork/>`_.
```

File: docs/source/contributing/contributing-guide.rst
Comment (RomainGoussault) [inline docs/source/contributing/contributing-guide.rst:54]:
```suggestion
Discussions about Substra take place on the repositories's Issues and Pull Requests sections and on Slack. Anybody is welcome to join these conversations.
```

Comment (RomainGoussault) [reply docs/source/contributing/contributing-guide.rst:54]: I would also add a link to the Community page.

**Review by EazyAl — APPROVED** [review]
Looks good to me :-)

File: CONTRIBUTORS.md
Comment (RomainGoussault) [inline CONTRIBUTORS.md:1]:
```suggestion
This is a file of people that have made significant contributions to the Substra documentation. It is sorted in chronological order. Please include your contribution at the bottom of this document in the following format: name (N), email (E), description of work (W) and date (D).
```

**Review by ThibaultFy — COMMENTED** [review]
LGTM, I'll let Romain approve it :)

File: docs/source/contributing/contributing-guide.rst
Comment (RomainGoussault) [inline docs/source/contributing/contributing-guide.rst:17]:
```suggestion
Contributors are any person that have contributed to the code. It does not matter whether it's a typo fix or 1000 lines of code. Making a contribution however does not automatically entitle you to copyright over that code. For copyright the contribution must be significant enough meet the `threshold of originality <https://en.wikipedia.org/wiki/Threshold_of_originality>`_, which basically means that your code is somewhat unique and non-generic. Fixing a typo does not give you access to copyright over that word or sentence.
```

File: docs/source/contributing/contributing-guide.rst
Comment (RomainGoussault) [inline docs/source/contributing/contributing-guide.rst:31]:
```suggestion
* `Fork the repository <https://guides.github.com/activities/forking/>`_ and clone it locally. Connect your local to the original "upstream" repository by adding it as a remote. Pull in changes from "upstream" often so that you stay up to date so that when you submit your pull request, merge conflicts will be less likely. See more detailed instructions `here <https://help.github.com/articles/syncing-a-fork/>`_.
```

File: docs/source/contributing/contributing-guide.rst
Comment (RomainGoussault) [inline docs/source/contributing/contributing-guide.rst:62]:
```suggestion
This guide follows guidelines from `opensource.guide <https://github.com/github/opensource.guide>`_.
```

**Review by RomainGoussault — APPROVED** [review]
Thanks for the PR, that's huge progress for improving contributors experience!

I just suggested some typos to fix, otherwise LGTM!

**Review by ThibaultFy — APPROVED** [review]

**Pull request merged**
